crates/hyperswitch_connectors/src/connectors/novalnet/transformers.rs

```diff
@@ -1064,10 +1085,90 @@ impl<F>
 }
 
 //TODO: Fill the struct with respective fields
-#[derive(Default, Debug, Serialize, Deserialize, PartialEq)]
+#[derive(Default, Debug, Serialize, Deserialize)]
 pub struct NovalnetErrorResponse {
-    pub status_code: u16,
+    pub status_code: u32,
     pub code: String,
     pub message: String,
     pub reason: Option<String>,
 }
+
+#[derive(Display, Debug, Serialize, Deserialize)]
+#[strum(serialize_all = "SCREAMING_SNAKE_CASE")]
+#[serde(rename_all = "SCREAMING_SNAKE_CASE")]
+pub enum WebhookEventType {
+    Payment,
+    TransactionCapture,
+    TransactionCancel,
+    TransactionRefund,
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+pub struct NovalnetWebhookEvent {
+    pub checksum: String,
+    pub tid: i64,
+    pub parent_tid: Option<i64>,
+    #[serde(rename = "type")]
+    pub event_type: WebhookEventType,
+}
+
+#[derive(Debug, Serialize, Deserialize, Clone)]
+#[serde(untagged)]
+pub enum NovalnetWebhookTransactionData {
+    OtherTransactionData(NovalnetSyncResponseTransactionData),
+    CaptureTransactionData(NovalnetCaptureTransactionData),
+    CancelTransactionData(NovalnetPaymentsResponseTransactionData),
+    RefundsTransactionData(NovalnetRefundsTransactionData),
+}
+#[derive(Serialize, Deserialize, Debug)]
+pub struct NovalnetWebhookNotificationResponse {
+    pub customer: Option<NovalnetResponseCustomer>,
```

Comment:
do we consume all the fields in response ? like customer from webhooks body, if not can you remove them?